### PR TITLE
Docs: Updates L2 Announce for LB Class Support

### LINKED ISSUE #28236: Docs: Add LoadBalancerClass to L2 Announcements
Now that https://github.com/cilium/cilium/pull/27976 merged, update the [l2-announcements](https://docs.cilium.io/en/stable/network/l2-announcements/#l2-announcements) docs for LoadBalancerClass support

### PR DESCRIPTION
**Pull request by danehans**
Updates L2 announcer docs for [load balancer class](https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class) support.

Fixes: #28236

### CODE REVIEW

**Review by dylandreimerink — APPROVED** [review]

File: Documentation/network/l2-announcements.rst [hunk 1]
```diff
@@ -114,7 +114,9 @@ Service Selector
 
 The service selector is a `label selector <https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/>`__ 
 that determines which services are selected by this policy. If no service 
-selector is provided, all services are selected by the policy.
+selector is provided, all services are selected by the policy. A service must have
+`loadBalancerClass <https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class>`__
+unspecified or set to "io.cilium/l2-announcer" to be selected by a policy for announcement.
```
Comment (learnitall) [inline Documentation/network/l2-announcements.rst]:
```suggestion
unspecified or set to ``io.cilium/l2-announcer`` to be selected by a policy for announcement.
```

**Review by learnitall — CHANGES_REQUESTED** [review]
One small nit on formatting, but otherwise LGTM.

File: Documentation/network/l2-announcements.rst (hunk 1 above)
Comment (danehans) [reply Documentation/network/l2-announcements.rst]: Commit 77d0747 resolves this comment.

**Review by learnitall — APPROVED** [review]

**Comment by danehans** [comment]
@dylandreimerink @learnitall do you have any suggetions for getting this PR merged?

**Comment by learnitall** [comment]
Hey @danehans I'll mark this `ready-to-merge` so it'll get picked up.

Docs-only change, full test suite is not required and all other tests are green. Marking ready to merge.

**Pull request merged**